doc/source/whatsnew/v1.2.0.rst

```diff
@@ -132,6 +133,7 @@ Plotting
 Groupby/resample/rolling
 ^^^^^^^^^^^^^^^^^^^^^^^^
 
+- :meth:`DataFrame.groupby` aggregations of categorical series will now return a :class:`Categorical` while preserving the codes and categories of the original series
```

Comment:
can you add this PR as the issue (or relevant issue if discussed elsewhere.

Reply:
Added the PR as the issue in commit 0fd3dde3f